fetch/client/src/test/scala/endpoints4s/fetch/future/BasicAuthenticationTestApi.scala

```diff
@@ -0,0 +1,19 @@
+package endpoints4s.fetch.future
+
+import endpoints4s.algebra
+import endpoints4s.algebra.EndpointsTestApi
+
+trait BasicAuthenticationTestApi extends EndpointsTestApi with algebra.BasicAuthentication {
```

Comment:
Don’t we already have a `TestApi` class that we should reuse here?

Reply:
Issue with previous client test implementation is that same endpoint is re-mocked with `WireMock` to produce a different result. Because I had to set up the mocks before launching tests, I needed new endpoints for each unique result.